### DISCUSSION TITLE
Creating a new Workspace results in "redirect_uri is not authorized"

### DISCUSSION THREAD

**Discussion by plpsanchez** [fief-dev/fief#55] [Q&A] [answered]
I would like to have a couple of Fief workspaces in a quickstarted self hosted domain running in the single container. Certainly the solution is very simple, my only problem is that I can not find it  :-) .

So I am trying to run Fief in a custom local domain and port and everything seems ok regarding the main workspace. Problem arises when I try to create another workspace via the top-right hand side menu. That results in a "redirect_uri is not authorized".

I can not use port 8000 in my host because it is being already used, so I tried to use port 8500 by means of docker run+env variables.
No matter how I changed PORT, ROOT_DOMAIN and FIEF_DOMAIN environment variables, and even the port mapping, the system does make acesses to the new port number but still tries to access port 8000 on localhost.

Digging into the code, I found a few places where there are constant strings containing "localhost:8000" where, I think, should be 
made reference to the config variable that constains the value for the port.

I had some difficulty to grasp the difference between ROOT_DOMAIN and FIEF_DOMAIN, it seems to me that a few more lines in the docs on how to quickstart dockerized Fief in a self hosted domain would help a lot.

Cheers.

**Comment by frankie567** [comment, marked as answer]
Hello @plpsanchez 👋  Welcome to Fief!

Could you give me the value you're using for `PORT`, `ROOT_DOMAIN` and `FIEF_DOMAIN`?

Basically, the difference between `ROOT_DOMAIN` and `FIEF_DOMAIN` is the following:

* `ROOT_DOMAIN` is the root domain (no subdomain) of all your workspaces. For example, it's `fief.dev` for the cloud version. In local self-hosting the default is `localhost:8000`. It'll be used to generate other workspaces subdomains, e.g., `example.fief.dev` or `example.localhost:8000`.
* `FIEF_DOMAIN` is the domain of your *main* Fief workspace. For example, it's `fief.fief.dev` for the cloud version. In local self-hosting, the default is `localhost:8000`.

The *main* Fief workspace controls the access to the admin dashboard of all your workspaces. When you create a new workspace, a redirect URI is added to the default client of the *main* Fief workspace. If everything goes well, you should have a new redirect URI added to this client which should look like: `http://example.localhost:8000/admin/api/auth/callback`.

**Reply by plpsanchez** [reply]
Hello @frankie567 

Thank you for your answer.

During my tests I was running Fief dockerized behind Traefik, so any incoming requests via the server port 8000  were routed to port 8000 of the container running Fief. Also, incoming requests to any subdomain of the form  *.fief.xxx.eu  via ports 80 and 443 were also routed to port 8000 of Fief container. So, anything coming thru port 8000 or for a fief subdomain got delivered to the Fief container.
My intention is to access the dashboard at fief.xxx.eu/admin/   and have workspaces like  yyy.fief.xxx.eu

The environment variables were:
```
PORT=8000
ROOT_DOMAIN=fief.xxx.eu:8000
FIEF_DOMAIN=fief.xxx.eu:8000
```
All of this seems to me to be correct. But I found I had mistaken definitions of the redirect_uri 's .
So, the issue is now solved.
